cmake/scripts/common/AddonHelpers.cmake

```diff
@@ -120,16 +120,20 @@ macro (build_addon target prefix libs)
             foreach(src_file ${USED_SOURCES})
               file(STRINGS ${src_file} BIN_ADDON_SRC_PARTS)
               foreach(loop_var ${BIN_ADDON_SRC_PARTS})
+                string(REGEX REPLACE ".*#include" "#include" loop_var "${loop_var}")
                 string(FIND "${loop_var}" "#include" matchres)
                 if("${matchres}" EQUAL 0)
                   string(REPLACE " " ";" loop_var "${loop_var}")
                   list(GET loop_var 1 include_name)
```

Comment:
This is wrong if `#include` is followed by more then once space.

Reply:
No, is correct for them is the `*` after `.`

Reply:
Tested with `#include  <kodi/addon-instance/Screensaver.h>` and it's not detected.

Reply:
Have it checked in cmake regex, unfortunately, I did not find a usable way directly.
It is possible to use regex groups in output (`\\1`), but unfortunately if the group is missing, it's a bit crazy,

Found some options with https://regex101.com unfortunately, but cmake does not support this.

With this request it is still not perfect, but at least now more is found.

Reply:
You implement this in a very complicated way, regex replace, find, replace, select n-th element and regex replace instead of a single regex match.

Reply:
Found a bit better way (last commit).

Old:
```cmake
                  string(REPLACE " " ";" loop_var "${loop_var}")
                  list(GET loop_var 1 include_name)
                  string(REGEX REPLACE "[<>\"]|kodi/" "" include_name "${include_name}")
```

New:
```cmake
                  string(REGEX REPLACE "#include|[ <>\"]|kodi/" "" include_name "${loop_var}")
```

Also handle this all the spaces between `#include` and `<` and only one line.

With the space there, unfortunately, one was removed within <>, but since it refers to Kodi header check, which never have spaces in it, I do not see it bad.